.github/workflows/semver-checks.yml

```diff
@@ -1,8 +1,12 @@
 on:   # yamllint disable-line rule:truthy
-  pull_request:
+  pull_request_target:
```

Comment:
I've just realized, this is vulnerable. A malicious "contributor" could update build scripts to extract the secrets and rustdoc has to run the build scripts to properly build the documentation.

I was thinking of chaining two jobs with different permissions and only sending `true` or `false` between them but I'm not sure if it's possible since this declaration is for the entire file.

Reply:
So, the only secret in the whole repo is my PR-creating token, which only exists to work around other Github permission garbage. But I guess, we shouldn't leak it because it would allow people to create PRs and issues under my name, which would be very confusing at best.

But -- in your attack model, doesn't this malicious contributor need to update the build scripts *and* get them merged into master, before they'll do anything?

Reply:
They can do a mini Jia Tan (xz backdoor) and just get one contribution in.
That will get him to run CI workflow automatically when running CI.
Then they extract the secrets with something like:

```yml
    - name: Run a multi-line script
      env:
        SECRET_NAME: A_FRUIT_NAME
      run: |
        echo "SECRET_NAME = $SECRET_NAME"
        echo "SECRET_NAME = ${{ env.SECRET_NAME }}"
        SECRET_VALUE=${{ secrets[env.SECRET_NAME] }}
        echo "SECRET_VALUE = $SECRET_VALUE"
```

Ref: https://stackoverflow.com/a/61272209

Reply:
> But I guess, we shouldn't leak it because it would allow people to create PRs and issues under my name, which would be very confusing at best.

Exactly, and very annoying.

> But -- in your attack model, doesn't this malicious contributor need to update the build scripts _and_ get them merged into master, before they'll do anything?

I mean `build.rs`. To run semver checks we need to run `rustdoc`. `rustdoc` has to run `build.rs` because `build.rs` may emit relevant `cfg`s. So if we run rustdoc on an untrusted PR we're running an untrusted `build.rs`.

@storopoli that yaml wouldn't work, the attacker would have to write it in Rust.